Patches/CheckGameEndPatch.cs

```diff
@@ -557,231 +538,44 @@ public bool CheckGameEndByLivingPlayers(out GameOverReason reason)
-                        reason = GameOverReason.ImpostorByKill;
-                        winner = CustomWinner.Huntsman;
-                        rl = CustomRoles.Huntsman;
-                    }
-                    else if (Necro == totalNKAlive)
-                    {
-                        reason = GameOverReason.ImpostorByKill;
-                        winner = CustomWinner.Necromancer;
-                        rl = CustomRoles.Necromancer;
-                    }
-                    else if (Pyro == totalNKAlive)
-                    {
-                        reason = GameOverReason.ImpostorByKill;
-                        winner = CustomWinner.Pyromaniac;
-                        rl = CustomRoles.Pyromaniac;
-                    }
-                    else if (Hex == totalNKAlive)
-                    {
-                        reason = GameOverReason.ImpostorByKill;
-                        winner = CustomWinner.HexMaster;
-                        rl = CustomRoles.HexMaster;
-                    }
-                    else if (Bandit == totalNKAlive)
-                    {
-                        reason = GameOverReason.ImpostorByKill;
-                        winner = CustomWinner.Bandit;
-                        rl = CustomRoles.Bandit;
-                    }
-                    else if (RR == totalNKAlive)
-                    {
-                        reason = GameOverReason.ImpostorByKill;
-                        winner = CustomWinner.RuthlessRomantic; 
-                    }
-                    else if (WW == totalNKAlive)
-                    {
-                        reason = GameOverReason.ImpostorByKill;
-                        winner = CustomWinner.Werewolf;
-                        rl = CustomRoles.Werewolf;
-                    }
-                    else if (Shr == totalNKAlive)
-                    {
-                        reason = GameOverReason.ImpostorByKill;
-                        winner = CustomWinner.Shroud;
-                        rl = CustomRoles.Shroud;
-                    }
-                    else if (Arso == totalNKAlive)
-                    {
-                        reason = GameOverReason.ImpostorByKill;
-                        winner = CustomWinner.Arsonist;
-                        rl = CustomRoles.Arsonist;
-                    }
-                    else if (Glitch == totalNKAlive)
+                    var winners = neutralRoleCounts.Where(kvp => kvp.Value == totalNKAlive).ToList();
+                    var winnnerLength = winners.Count;
+                    if (winnnerLength == 1)
                     {
+                        var winnerRole = winners[0].Key.GetNeutralCustomRoleFromCountType();
                         reason = GameOverReason.ImpostorByKill;
-                        winner = CustomWinner.Glitch;
-                        rl = CustomRoles.Glitch;
+                        CustomWinnerHolder.ResetAndSetWinner(CustomRolesHelper.GetNeutralCustomWinnerFromRole(winnerRole));
```

Comment:
Use `CustomWinnerHolder.ResetAndSetWinner(winnerRole.GetNeutralCustomWinnerFromRole())` because the function has `this` method

Reply:
Also add `using TOHE.Modules;`